tests/test_tenant_unauthorized.py

```diff
@@ -1,49 +1,436 @@
+from dataall.modules.mlstudio.services.mlstudio_permissions import GET_SGMSTUDIO_USER, MANAGE_SGMSTUDIO_USERS
+from dataall.modules.notebooks.services.notebook_permissions import GET_NOTEBOOK, MANAGE_NOTEBOOKS
+from dataall.modules.omics.services.omics_permissions import MANAGE_OMICS_RUNS
+from dataall.modules.redshift_datasets.services.redshift_dataset_permissions import MANAGE_REDSHIFT_DATASETS
+from dataall.modules.s3_datasets.services.dataset_permissions import GET_DATASET, GET_DATASET_TABLE, MANAGE_DATASETS
+from dataall.modules.shares_base.services.share_permissions import MANAGE_SHARES
+from dataall.modules.worksheets.services.worksheet_permissions import MANAGE_WORKSHEETS
 
 
-load_modules(modes={ImportMode.API})
-
-OPT_OUT_MUTATIONS = {
-    'Mutation.updateGroupTenantPermissions': 'admin action. No need for tenant permission check',
-    'Mutation.updateSSMParameter': 'admin action. No need for tenant permission check',
-    'Mutation.createQuicksightDataSourceSet': 'admin action. No need for tenant permission check',
-    'Mutation.startMaintenanceWindow': 'admin action. No need for tenant permission check',
-    'Mutation.stopMaintenanceWindow': 'admin action. No need for tenant permission check',
-    'Mutation.startReindexCatalog': 'admin action. No need for tenant permission check',
-    'Mutation.markNotificationAsRead': 'tenant permissions do not apply to support notifications',
-    'Mutation.deleteNotification': 'tenant permissions do not apply to support notifications',
-    'Mutation.postFeedMessage': 'tenant permissions do not apply to support feed messages',
-    'Mutation.upVote': 'tenant permissions do not apply to support votes',
-    'Mutation.createAttachedMetadataForm': 'outside of this PR to be able to backport to v2.6.2',
-    'Mutation.deleteAttachedMetadataForm': 'outside of this PR to be able to backport to v2.6.2',
-    'Mutation.createRedshiftConnection': 'outside of this PR to be able to backport to v2.6.2',
-    'Mutation.deleteRedshiftConnection': 'outside of this PR to be able to backport to v2.6.2',
-    'Mutation.addConnectionGroupPermission': 'outside of this PR to be able to backport to v2.6.2',
-    'Mutation.deleteConnectionGroupPermission': 'outside of this PR to be able to backport to v2.6.2',
-}
+def resolver_id(type_name, field_name):
+    return f'{type_name}_{field_name}'
+
+
+class IgnoreReason(Enum):
+    ADMIN = 'admin action. No need for tenant permission check'
+    SUPPORT = 'permissions do not apply to support notifications'
+    FEED = 'permissions do not apply to support feed messages'
+    VOTES = 'permissions do not apply to support votes'
+    BACKPORT = 'outside of this PR to be able to backport to v2.6.2'
+    INTRAMODULE = 'returns intra-module data'
+    PERMCHECK = 'checks user permissions for a particular feature'
+    CATALOG = 'catalog resources are public by design'
+    SIMPLIFIED = 'simplified response'
+    NOTREQUIRED = 'permission check is not required'
+
```

Comment:
I really like that we specify IgnoreReasons! But I think we need to add some and maybe we can merge some together:

- `ADMIN` - all good! I would maybe rename it as `TENANT`
- `APPSUPPORT - permissions do not apply to application support features` - merging SUPPORT + FEED+ VOTES
- `BACKPORT`, `INTRAMODULE`
- `PERMCHECK` ---> it is very generic, we just want to account for the USERROLEINRESOURCE checks right?
- `PUBLIC` - Public by design  - It will be used in other places not only for Catalog
- ....

Reply:
TestData should define either permissions or the ignore reason, right now we are falling back to default NOTREQUIRED. Could we instead have some logic that throws an error for not-correctly defined TestData classes?

```
@dataclass
class TestData:
    tenant_perm: Optional[str] = None
    tenant_ignore: Optional['IgnoreReason'] = None
    resource_perm: Optional[str] = None 
    resource_ignore: Optional['IgnoreReason'] = None

    def __post_init__(self):
        # Validate that at least one of resource_perm or resource_ignore is set
        if self.resource_perm is None and self.resource_ignore is None:
            raise ValueError("Either resource_perm or resource_ignore must be set")
        if self.tenant_perm is None and self.tenant_ignore is None:
            raise ValueError("Either tenant_perm or tenant_ignore must be set")


```